File: app/src/main/java/com/github/overpass/gather/auth/register/add/ImageSourceUseCase.java
package com.github.overpass.gather.auth.register.add;

import androidx.lifecycle.LiveData;

import com.github.overpass.gather.SingleLiveEvent;

public class ImageSourceUseCase {

    private final SingleLiveEvent<ImageSource> imageSourceData;

    public ImageSourceUseCase() {
        imageSourceData = new SingleLiveEvent<>();
    }

    public LiveData<ImageSource> getImageSourceData() {
        return imageSourceData;
    }

    public void onGallery() {
        imageSourceData.setValue(ImageSource.GALLERY);
    }

    public void onCamera() {
        imageSourceData.setValue(ImageSource.CAMERA);
    }
}

File: app/src/main/java/by/overpass/gather/model/usecase/image/ImageSourceUseCase.kt
package by.overpass.gather.model.usecase.image

import androidx.lifecycle.LiveData
import by.overpass.gather.ui.auth.register.add.ImageSource
import com.hadilq.liveevent.LiveEvent

class ImageSourceUseCase constructor(
        private val imageSourceData: LiveEvent<ImageSource> = LiveEvent()
) {

    fun getImageSourceData(): LiveData<ImageSource> {
        return imageSourceData
    }

    fun onGallery() {
        imageSourceData.value = ImageSource.GALLERY
    }

    fun onCamera() {
        imageSourceData.value = ImageSource.CAMERA
    }
}
